plugins/calibre.koplugin/wireless.lua

```diff
@@ -234,39 +230,39 @@ function CalibreWireless:connect()
     end
 
     self.connect_message = false
-    local host, port
+    local address_type, host, port, ok, err
     if G_reader_settings:hasNot("calibre_wireless_url") then
         host, port = self:find_calibre_server()
+        if host and port then
+            address_type = "discovered"
+        else
+            ok = false
+            err = "Couldn't discover a calibre instance on the local network"
+            address_type = "unavailable"
+        end
     else
         local calibre_url = G_reader_settings:readSetting("calibre_wireless_url")
         host, port = calibre_url["address"], calibre_url["port"]
-        if not self:checkCalibreServer(host, port) then
-            host = nil
-        else
-            self.failed_connect_callback = function()
-                UIManager:show(InfoMessage:new{
-                    text = _("Cannot connect to calibre server."),
-                })
-                self:disconnect()
-            end
-            -- wait 10 seconds to connect to calibre
-            UIManager:scheduleIn(10, self.failed_connect_callback)
-        end
+        address_type = "specified"
     end
+
     if host and port then
+        ok, err = self:checkCalibreServer(host, port)
+    end
+
+    if not ok then
+        logger.warn(string.format("Cannot connect to %s calibre server at %s:%s (%s)", address_type, host or "????", port or "??", err or "N/A"))
```

Comment:
Maybe we should just show that to the user?

Reply:
I waffled a bit on that one. There's more context in the logs, and the error, if any, is very very very likely to be "connection refused", which doesn't really help much.

That leaves the IP/port pair as potentially always useful and understandable info, at least?

Reply:
There would also be localization issues, because that's essentially a low-level `socket` error, untranslated.

Reply:
The point is more that the user can just snap a quick pic of their screen without having to dig in logs. I don't think the errors need to be super localized necessarily (though ideally, sure). :-)

Reply:
Fair enough ;).